.github/workflows/maven.yaml

```diff
@@ -84,7 +109,7 @@ jobs:
           KROXYLICIOUS_KMS_FORTANIX_API_ENDPOINT: ${{ vars.KROXYLICIOUS_KMS_FORTANIX_API_ENDPOINT }}
           KROXYLICIOUS_KMS_FORTANIX_ADMIN_API_KEY: ${{ secrets.KROXYLICIOUS_KMS_FORTANIX_ADMIN_API_KEY }}
           KROXYLICIOUS_KMS_FORTANIX_API_KEY: ${{ secrets.KROXYLICIOUS_KMS_FORTANIX_API_KEY }}
-        run: mvn -B clean verify -Pci -Djapicmp.skip=${REFERENCE_RELEASE_UNPUBLISHED} org.sonarsource.scanner.maven:sonar-maven-plugin:3.9.1.2184:sonar -Dsonar.projectKey=kroxylicious_kroxylicious
+        run: mvn -B verify -Pci -Djapicmp.skip=${REFERENCE_RELEASE_UNPUBLISHED} org.sonarsource.scanner.maven:sonar-maven-plugin:5.0.0.4389:sonar -Dsonar.projectKey=kroxylicious_kroxylicious -DskipITs
```

Comment:
I don't think we want to skipITs.  We still want to run all tests on a push to main.

Reply:
In fact we should refactor "Build Kroxylicious maven project" and "Build Kroxylicious maven project" so they are a single build step with some bash conditionality which injects the additional arguments into the maven command line if  SONAR_TOKEN_SET == 'true'..

Reply:
My intent was to run all steps on push but only run ITs once we had an image. 

So build with unit tests for a “quick” sanity check the image is worth building then run the image build (so in theory everything should be in m2 cache) then run the ITs.

Reply:
Ah sorry, there is no coverage recorded for sonar so ITs don’t help, but realise now that we aren’t running a separate sonar run here it’s done inline so yes we should remove the skipITs flag there. Thanks for catching